asdpc-2024-issuance.md

```diff
@@ -0,0 +1,168 @@
+users of the oracle for shorter latency.
+
+Because the transactions that are involved in the process of commitment, reveal and tally of oracle queries require
+confirmation by inclusion in a block, in the best scenario, the time-to-resolution of an oracle query can be no less 
+than the time that it takes to append 4 blocks to the tip of the chain (1 for publishing the query itself, 1 for 
+commitments, 1 for reveals, and finally 1 for the tally.
+
+With a block time of 45 seconds, those 4 blocks equal to 3 minutes. While this latency may be perfectly fine for many
+use cases, there are some of those in which the freshness of the data being attested is key. Any reduction to this 
+latency—even if marginal—can make a big difference to them.
+
+### Considered and alternative solutions
+
+[WIP-0003] itself explores several alternative coin emission schedules. However, they are all based on the same 
+Bitcoin-like model with asymptotic inflation and halvings.
+
+The discussion on improving the coin emission schedule has been going on for many years, but the matter started to 
+receive more attention as soon as Proof of Stake started to shape up as the consensus algorithm for wit/2.
+
+Over the last year, other proposals like reducing the block rewards in shorter periods and milder installments have 
+been brought to the table, but the issue of the asymptotic convergence to zero of the inflation was not addressed by 
+them.
+
+During the process of drafting this document, several issuance schedules focusing on modulating and eventually fixing
+the inflation instead of the quantity of the block rewards were explored. These looked very promising for 
+fulfilling the goal of long term sustainability. However, we found that the balance between providing attractive 
+incentives for validators and keeping the total coin supply contained needs to be adjusted very carefully. Therefore,
+we reached the conclusion that an eventual transition to a fixed inflation schedule—while still strictly needed—will 
+require further examination of the reality of the network and the distribution of stake once wit/2 is rolled out.
+
+On the subject of the block time, aka _epoch period_, there have not been any significant discussions on the community 
+channels lately. However, several past and recent experiments performed by node operators and relevant contributors to 
+the design and development of the protocol have shown that a block time shorter than 15 seconds can jeopardize the 
+stability of the network, but a 20-30 seconds block time allows the protocol to perform just as well as with the current
+45 seconds block time.
+
+## Specification
+
+### Calculation of block rewards
+
+**()** The block reward MUST NOT be calculated any longer using the old function that computed it for every block based 
+on an initial reward, the current protocol epoch, and a halving period.
+
+**()** The block reward MUST be fixed to `125` Wit.
```

Comment:
@guidiaz @Tommytrg @parodyBit what do you think?

Reply:
My proposal would be a 20s block time and 50 wit reward per block.

Reply:
> My proposal would be a 20s block time and 50 wit reward per block.

Why so?

Reply:
The halvening on the current version of the network is actually still [more than a year away](https://github.com/witnet/witnet-rust/blob/4f3f56b03c0dabc4f1d3fdbf7b25ed4dd77d2a91/config/src/defaults.rs#L178). 

The reason why I think a reduction in block reward makes sense is that it will become much more profitable to run a node as soon as you stake a (yet-to-be-seen) amount of WIT compared to now (since you now need to run x nodes to utilize the same amount of WIT in an optimal way). Hence, we would (significantly) overpay to achieve the same level of security of the network (compared to now). I think we can draw an easy parallel here with how ETH rewards were reduced when they moved to PoS and I don't see why we should not follow the same paradigm.

As far as the actual numbers are concerned, yes, both of those are picked for the sake of easy-to-calculate integer values.

Reply:
I agree on the idea that a high yearly inflation rate leads to "excess tokens (that) get dump which leads to a lower coin price which leads to lower dollar value staked on validators", though. 

I don't agree with the reasoning that a reduction in block rewards makes running a node much more profitable. On the contrary, the less yearly inflation the less profit (in terms of $WIT, ofc). The improvement that you describe concerning node operators being able to achieve the same level of network security by running less nodes is true, but it comes because of the new PoS mechanism, and not because a reduction in block reward is being proposed.

Nevertheless, I do agree that:

- reducing inflation to anything around 50% would be appropriate and pretty convenient to all stakeholders (at least in the long term);
- it is the Wit/2 upgrade launch the right and only moment to introduce a reduction in the block time (good for developers, supported dapps and data requesters) and an overall reduction of the yearly inflation rate (good for miners/validators, the overall network security and the tokenomics). 
- the final block reward should be fine-tuned before release of the Wit/2 upgrade on the Witnet protocol depending on the minimum block time possible reached during the beta testing of a Wit/2 upgraded testnet.

Reply:
> I don't agree with the reasoning that a reduction in block rewards makes running a node much more profitable, though. On the contrary, the less yearly inflation the less profit (in terms of $WIT, ofc). The improvement that you describe concerning node operators being able to achieve the same level of network security by running less nodes is true, but it comes because of the new PoS mechanism, and not because a reduction in block reward is being proposed.

You misunderstood me, I was not implying that a reduction in rewards will result in an increase in profitability. I was simply saying that the release of wit\2 will make running a node much more profitable because you will need much less hardware to optimally use a fixed amount of WIT compared to the current state. Hence, a reduction makes sense (for many more reasons).